recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh

```diff
@@ -479,13 +485,112 @@ function is_installed() {
     fi
 }
 
+#
+# A helper function that compares two software versions.
+#
+# version number schema:
+# - based on Major.Minor.Patch.Buildnumber
+# - based only on digits
+#
+# If version1 newer than version2           -> return 1
+# If version1 older than version2           -> return 2
+# If version1 equal version2                -> return 3
+# Illegal format in version1 or version2    -> return 0
+#
+# Usage: version_comp $version1 $version2
+#
+# shellcheck disable=SC2034
+function version_comp () {
+
+    local regex='^([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)$'
+    local -a v1
+    local -a v2
+
+    if [[ "$1" =~ $regex ]]
+    then
+        v1=("${BASH_REMATCH[@]}")
+        if [[ "$2" =~ $regex ]]
+        then
+            v2=("${BASH_REMATCH[@]}")
+        else
+            return 0    # version format illegal
+        fi
+    else
+        return 0        # version format illegal
+    fi
```

Comment:
better check v1 and then v2 on the same level. no need to check v2 in if of v1 check

Reply:
since you don't echo an error you could also have only one if where checking both, v1 and v2

Reply:
changed as discussed